### PR TITLE
Add TSP header file

### PR DESCRIPTION
**Pull request by jyao1**
Add CXL 3.1 TEE Security Protocol (TSP) header file.

Only mandatory commands are included in this patch.
The optional commands will be added later.

### CODE REVIEW

File: include/industry_standard/cxl_tsp.h [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+/* only valid in CAP, not in SET/GET */
+#define CXL_TSP_MEMORY_ENCRYPTION_FEATURES_SUPPORT_INITIATOR_SUPPLIED_ENTROPY 0x8
+/* offset is changed in SET/GET */
+#define CXL_TSP_MEMORY_ENCRYPTION_FEATURES_SUPPORT_CKID_BASED_REQUIRED 0x10
+
+#define CXL_TSP_MEMORY_ENCRYPTION_ALGORITHMS_AES_XTS_128 0x1
+#define CXL_TSP_MEMORY_ENCRYPTION_ALGORITHMS_AES_XTS_256 0x2
+#define CXL_TSP_MEMORY_ENCRYPTION_ALGORITHMS_VENDOR_SPECIFIC 0x80000000
+
+#define CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_WRITE_ACCESS_CONTROL 0x1
```
Comment (steven-bellock) [inline include/industry_standard/cxl_tsp.h:99]: These names are too long. Maybe go with `CONTROL` -> `CTRL`, `FEATURES` -> `FEAT`, etc.

File: include/industry_standard/cxl_tsp.h [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+    CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_EXPLICIT_OUT_OF_BAND_TE_STATE_CHANGE \
+    0x8
+#define \
+    CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_EXPLICIT_IN_BAND_TE_STATE_CHANGE \
+    0x10
+#define \
+    CXL_TSP_TE_STATE_CHANGE_AND_ACCESS_CONTROL_FEATURES_SUPPORT_EXPLICIT_TE_STATE_CHANGE_SANITIZE \
+    0x20
+
+#define CXL_TSP_EXPLICIT_OUT_OF_BAND_TE_STATE_CHANGE_GRANULARITY_64B 0x1
```
Comment (steven-bellock) [inline include/industry_standard/cxl_tsp.h:112]: Any single bit is legal. If these are going to be `#define`d it should include all of them.

File: include/industry_standard/cxl_tsp.h [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+/* TSP error code */
+
+#define CXL_TSP_ERROR_CODE_INVALID_REQUEST 0x01
+#define CXL_TSP_ERROR_CODE_BUSY 0x02
+#define CXL_TSP_ERROR_CODE_UNSPECIFIED 0x03
+#define CXL_TSP_ERROR_CODE_UNSUPPORTED_REQUEST 0x04
+#define CXL_TSP_ERROR_CODE_VERSION_MISMATCH 0x05
+#define CXL_TSP_ERROR_CODE_VENDOR_SPECIFIC_ERROR 0x06
+#define CXL_TSP_ERROR_CODE_NO_PRIVILEDGE 0x07
```
Comment (steven-bellock) [inline include/industry_standard/cxl_tsp.h:296]: `PRIVILEGE`

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: include/industry_standard/cxl_tsp.h (hunk 2 above)
Comment (jyao1) [reply include/industry_standard/cxl_tsp.h:112]: done.

File: include/industry_standard/cxl_tsp.h (hunk 3 above)
Comment (jyao1) [reply include/industry_standard/cxl_tsp.h]: fixed

File: include/industry_standard/cxl_tsp.h (hunk 1 above)
Comment (jyao1) [reply include/industry_standard/cxl_tsp.h]: I reduced the length.

**Review by steven-bellock — APPROVED** [review]

**Pull request merged**